crates/bevy_utils/src/label.rs

```diff
@@ -1,9 +1,15 @@
 //! Traits used by label implementations
 
-use std::{
-    any::Any,
-    hash::{Hash, Hasher},
-};
+use std::any::{Any, TypeId};
+use std::collections::hash_map::DefaultHasher;
+use std::fmt::Debug;
+use std::hash::{Hash, Hasher};
+use std::sync::{OnceLock, RwLock};
+
+use crate::{default, HashMap};
+
+type Interner = RwLock<HashMap<UniqueValue, &'static str>>;
+static INTERNER: OnceLock<Interner> = OnceLock::new();
```

Comment:
As far as I can tell, you only ever call `.write()` on this, so it would be better to just use a mutex here.

Reply:
I think trading the `RwLock` for a `Mutex` could end up causing some weird behavior in hypothetical situations where systems or sub-apps in different threads try to construct or run schedules at the same time.

Reply:
What I mean is, doesn't that prevent different threads from reading the interned strings at the same time?

Reply:
A `RwLock` only allows multiple threads to read it at once if you use `.read()`. Since you're only using `.write()`, it behaves like a mutex and only one thread can access the interner at once.

In order to allow multiple threads to read it at once you'd need to refactor this to only call `.write()` when a string is not already interned.

```rust
if let Some(str) = INTERNER.read().unwrap().get(key) {
    *str
} else {
    let str: &'static str = // format and leak the value
    INTERNER.write().unwrap().insert(key, str);
    str
}
```

Reply:
Ah, I see. It went over my head what you meant by "not using it like a `RwLock`".